app/src/main/java/ch/hikemate/app/model/authentication/AuthViewModel.kt

```diff
@@ -10,7 +10,7 @@ import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 
-class AuthViewModel(private val repository: AuthRepository) : ViewModel() {
+class AuthViewModel(private val repository: FirebaseAuthRepository) {
```

Comment:
Warning: I think you missed merge this. You should keep the Interface version (with the `ViewModel()`)